### DISCUSSION TITLE
Fetch external data for GraphQL API

### DISCUSSION THREAD

**Discussion by ekafyi** [prisma/prisma#9451] [Q&A] [answered]
Hi all, I just learned to make a GraphQL API with Prisma + Nexus from this starter https://github.com/prisma/prisma-examples/tree/latest/typescript/graphql . The sample code was perfect, clear and easy to modify & extend. 🙌🏽

However, I'm not sure what's the best way to _fetch data from an external REST endpoint_ and _store it in the DB_.

Should I simply run everything in the Mutation resolver in the schema (eg. here https://github.com/prisma/prisma-examples/blob/latest/typescript/graphql/src/schema.ts#L49 ), or is there a better, recommended way?

On vanilla Apollo Server, for instance, there is a `RESTDataSource` class that could be used for similar purpose: https://www.apollographql.com/docs/tutorial/data-source/#connect-a-rest-api

Appreciate any pointer.

**Comment by ryands17** [comment, marked as answer]
@ekafyi 👋 

A `RESTDataSource` should go well in this case if you're using Apollo. It's simple to work with and you can easily write the `Query`/`Mutation` needed to make this work by calling the datasource API.

**Reply by ekafyi** [reply]
Thank you for the answer! So there is no "blessed" Prisma way to do this? If I'm using Prisma with Nexus and passing the schema to the Apollo Server instance (not defining resolvers the Apollo way), I can still use Apollo's RESTDataSource?

**Reply by ryands17** [reply]
Yes you can. Prisma is layer agnostic and has no relation to GraphQL so you should be able to add a separate Data Source to Apollo.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
